pkg/build/api/v1/types.go

```diff
@@ -176,8 +176,46 @@ type BuildStatus struct {
 
 	// output describes the Docker image the build has produced.
 	Output BuildStatusOutput `json:"output,omitempty" protobuf:"bytes,10,opt,name=output"`
+
+	// steps contains details about each build step.
+	Steps Steps `json:"steps,omitempty" protobuf:"bytes,11,opt,name=steps"`
```

Comment:
I can confirm the stage flow nodes are the steps.

The nodes also have an ID and a parent node. Parent node might be important if we want to tell if steps are parallel vs sequential (although I'd need to look at this more).

There is some doc on the JSON here:

https://github.com/jenkinsci/pipeline-stage-view-plugin/tree/master/rest-api